### ISSUE TITLE
Custom query generation breaks if I include a where part

### ISSUE THREAD

**Issue by Dev-Owl** [simolus3/moor#296]
Hi,
I looked into the custom query part using moor files. Given the following code:
```dart

//Used as the base model for all tabels aka data contract
class BaseModel extends Table {
  TextColumn get id => text()();
  DateTimeColumn get lastchange => dateTime()();
  TextColumn get lastChangeBy => text()();
  BoolColumn get isDeleted => boolean().withDefault(Constant(false))();

  @override
  Set<Column> get primaryKey => {id};
}

class Users extends BaseModel {
  TextColumn get firstName =>
      text().withLength(min: 3).withDefault(const Constant("Max"))();
  TextColumn get lastName =>
      text().withLength(min: 3).withDefault(const Constant("Power"))();
  TextColumn get email => text()();
  TextColumn get position => text()();
}

@UseMoor(
  tables: [
    Users,
  ],
  include:{'queries.moor'}
)
class DbContext extends _$DbContext {
  DbContext()
      : super(FlutterQueryExecutor.inDatabaseFolder(path: 'backtrack.sqlite'));

  @override
  int get schemaVersion => 1;
}
```
And this simple moor file:
```dart
import 'dbContext.dart';

countAllUser: SELECT COUNT(*) FROM users WHERE isDeleted=0;
```

When I run `flutter packages pub run build_runner build` it breaks as soon as my query contains any **WHERE** condition with the following output:
```
[SEVERE] moor_generator:moor_generator on lib/data/dbContext.dart:
Error running MoorGenerator
Bad state: Unknown column null
[SEVERE] moor_generator:moor_generator on lib/data/dbContext.dart:
Error running DaoGenerator
Bad state: Unknown column null
```
If I remove the **WHERE** part of the query the code is generated, I'm sure its me but even after an hour I don't see what I did wrong. I need a soft delete function and of course it needs to be checked for each query to exclude the deleted data.

**Comment by North101** [comment]
So I know that when declaring in SQL it'll convert snake_case to camelCase but i'm not sure if it also does it the other way around.

Try doing `is_deleted = 0` or 
```
  BoolColumn get isDeleted => boolean().named('isDeleted').withDefault(Constant(false))();

```

**Comment by Dev-Owl** [comment]
You are right, I just discovered this on my own by reviewing the code that was generated but you were quicker. This is the line from the generated code that made "aha!" moment for me:

`GeneratedBoolColumn('is_deleted', $tableName, false,
        defaultValue: Constant(false));`

**Comment by simolus3** [comment]
Thanks for the report, and sorry for the inconvenience. Moor converted camelCase getters in Dart to snake_case columns since its first version. That didn't really matter at first, since the sql parser was added in a later version. Changing that behavior now (or even back then) would be breaking, so it's unlikely to happen. 

This is still a surprising behavior, and the `Bad state: Unknown column null` error is terrible UX from the generator. I've improved stability and error handling recently, so in the next moor version this query would fail with:
```
line 3, column 48: Unknown column. These columns are available: id in users, lastchange in users, last_change_by in users, is_deleted in users, first_name in users, last_name in users, email in users, position in users
  ╷
3 │ countAllUser: SELECT COUNT(*) FROM users WHERE isDeleted=0;
  │                                                ^^^^^^^^^
  ╵
```

I hope that makes it easier to understand what's going wrong.
